### ISSUE TITLE
LDAP Authentication is on the roadmap?

### ISSUE THREAD

**Issue by evertton** [docmost/docmost#118]
Is there a plan to implement LDAP authentication in future versions?

**Comment by dr1ss** [comment]
If it's not, it should be. LDAP / OIDC is a must have for this kind of applications.

**Comment by ados8** [comment]
LDAP and OIDC are completely different.
If an external authentication method is implemented I would vote for OIDC/OpenID.
Minimise development burden and platforms such as Authentik or Authelia can bridge the gap for those that need LDAP.

**Comment by k8ieone** [comment]
I agree, OIDC is more versatile and there are OIDC -> LDAP connectors for those that need a connection to LDAP.

**Comment by evertton** [comment]
About OIDC that is a topic [#22](https://github.com/docmost/docmost/issues/22).

**Comment by ados8** [comment]
> About OIDC that is a topic [#22](https://github.com/docmost/docmost/issues/22).

Hopefully OIDC is implemented in place of LDAP.
Minimise development burden and use an SSO app to bridge the gap for those that need LDAP.

**Comment by tiredofit** [comment]
LDAP and OIDC are not one the same.

While they both provide a mechanism for access into the application there are significant differences..

- With OIDC, you can't prepopulate your user directory and perform synchornization of user objects on a scheduled basis. Instead it requires a user to "login". This makes things challenging for provisioning (on/offboarding).

- With LDAP - You can't "auto login" - This is where OIDC comes in via a centralized authentication provider.

- Both allow abilities to export attributes (think name, address, phone number) and also things like roles or group memberships. LDAP has more flexibility in terms of configuring the application within the application and from your OIDC identity provider you have the ability to ETL your various bits to send to the application.

I'd be happy with both options, but if there was only one I would choose LDAP with the ability to login via a HTTP header. Far more flexibility.

**Comment by Shadowfita** [comment]
@Philipinho would you accept a PR that implements the windows NTLM authentication protocol that pulls additional AD information via LDAP? This will allow automatic user authentication and identification without needing to prompt for any username/email or password.

**Comment by Philipinho** [comment]
@Shadowfita not yet. I shared my thoughts here https://github.com/docmost/docmost/issues/22#issuecomment-2211018048.

**Comment by Shadowfita** [comment]
Thanks @Philipinho. If that is the case, I will maintain a downstream branch with NTLM v1 authentication. Docmost is something I am hoping to use in our workplace, and this is a requirement for us!

Once it's working, I will open a PR but leave it in draft so that you can at least review should you wish.

**Comment by ajaxovic** [comment]
For companies, SSO is must have. MR from @Shadowfita came nad I home it will be included. Otherwise, we can't use this. It is little bit sad, because rest of the project looks great.

**Comment by Philipinho** [comment]
LDAP is available in the Enterprise edition.
You can contact sales@docmost.com to purchase an enterprise license.
